keysManagement/keysHolder.go

```diff
@@ -42,14 +45,17 @@ func NewVirtualPeersHolder(args ArgsVirtualPeersHolder) (*virtualPeersHolder, er
 		return nil, err
 	}
 
-	return &virtualPeersHolder{
+	holder := &virtualPeersHolder{
 		data:                             make(map[string]*peerInfo),
 		pids:                             make(map[core.PeerID]struct{}),
 		keyGenerator:                     args.KeyGenerator,
 		p2pIdentityGenerator:             args.P2PIdentityGenerator,
 		isMainMachine:                    args.IsMainMachine,
 		maxRoundsWithoutReceivedMessages: args.MaxRoundsWithoutReceivedMessages,
-	}, nil
+		namedIdentitiesMap:               createNamedIdentitiesMap(args.NamedIdentities),
```

Comment:
we should not have a different map for this: instead, place all the data in the data member of type make(map[string]*peerInfo), add new fields in the peerInfo struct
If we do not found a provided named identity for a certain BLS pk, take the ones provided in the prefs.toml L8 & L11